icu4j/main/classes/core/src/com/ibm/icu/text/DateTimePatternGenerator.java

```diff
@@ -213,40 +213,25 @@ private void consumeShortTimePattern(String shortTimePattern, PatternInfo return
     private class AppendItemFormatsSink extends UResource.Sink {
         @Override
         public void put(UResource.Key key, UResource.Value value, boolean noFallback) {
-            UResource.Table itemsTable = value.getTable();
-            for (int i = 0; itemsTable.getKeyAndValue(i, key, value); ++i) {
-                int field = getAppendFormatNumber(key);
-                assert field != -1;
-                if (getAppendItemFormat(field) == null) {
-                    setAppendItemFormat(field, value.toString());
-                }
+            int field = getAppendFormatNumber(key);
+            assert field != -1;
```

Comment:
Pre-existing issue, but still...

The C++ code skips/ignores unknown keys, while this Java code here asserts that the key is known. Which means that when assertions are disabled (which is the default), then we ram forward anyway; unclear if the next line compensates for that. And when assertions are enabled, then we crash and burn.

We probably want to skip like in C++ (and like in the `put()` below): `if (field < 0) { return; }`

(We could make it strict and fail, but I generally prefer data readers to be lenient with additional data, as a way of being somewhat forward compatible.)

Reply:
I agree.  I replaced the assert with `if (field < 0) { return; }`.